### DISCUSSION TITLE
How to use async/await with mongoose.createConnection() ?

### DISCUSSION THREAD

**Discussion by chriscodes91** [Automattic/mongoose#12051] [Q&A] [answered]
I've always used mongoose.connect() with async/await for querying and such. 
Recently found myself needing to use mongoose.createConnection, however, it seems async/await doesn't work with it, only callbacks. 
Which is fine, but I really prefer async/await. Is there something I'm missing?

Thanks!

**Comment by Uzlopak** [comment, marked as answer]
https://mongoosejs.com/docs/api.html#mongoose_Mongoose-createConnection

> Returns:
> «Connection» the created Connection object. Connections are not thenable, so you can't do `await mongoose.createConnection()`. To await use mongoose.createConnection(uri).asPromise() instead.
>

**Reply by chriscodes91** [reply]
Thanks for the response. Actually didn't catch that in the docs.
But, I still can't get await to work on actual model queries to mongo.
I should have been a bit clearer, for example, this is what I want to accomplish.

`const conn = mongoose.createConnection('mongodb://localhost:27017/db')`
`const SomeModel = conn.model('SomeModel', require('./models/somemodel'))`
`const query = await SomeModel.findOne({_id: 1})` // I want to get this to work with await, instead of .then

**Reply by Uzlopak** [reply]
```javascript
const conn = await mongoose.createConnection('mongodb://localhost:27017/db').asPromise()
const SomeModel = conn.model('SomeModel', require('./models/somemodel'))
const query = await SomeModel.findOne({_id: 1})
```
?

**Reply by chriscodes91** [reply]
Well, thats embarrasing, that worked!
I actually added the asPromise() just didn't save I guess 🙃. Thanks!!
